src/engine/source/api/src/test/handlers.cpp

```diff
@@ -458,6 +464,117 @@ api::Handler sessionsDelete(shared_ptr<Router> router, shared_ptr<Catalog> catal
     };
 }
 
+api::Handler runPost(std::shared_ptr<::router::Router> router)
+{
+    return [router](api::wpRequest wRequest) -> api::wpResponse
+    {
+        using RequestType = eTest::RunPost_Request;
+        using ResponseType = eTest::RunPost_Response;
+        auto res = ::api::adapter::fromWazuhRequest<RequestType, ResponseType>(wRequest);
+
+        // If the request is not valid, return the error
+        if (std::holds_alternative<api::wpResponse>(res))
+        {
+            return std::move(std::get<api::wpResponse>(res));
+        }
+
+        // Validate the params request
+        auto& eRequest = std::get<RequestType>(res);
+        auto errorMsg = !eRequest.has_name()    ? std::make_optional("Missing /session name")
+                        : !eRequest.has_event() ? std::make_optional("Missing /event")
+                                                : std::nullopt;
+
+        if (!eRequest.has_debugmode())
+        {
+            eRequest.set_debugmode(eTest::DebugMode::ONLY_OUTPUT);
+        }
+
+        if (!eRequest.has_protocolqueue())
+        {
+            eRequest.set_protocolqueue(TEST_DEFAULT_PROTOCOL_QUEUE);
+        }
+
+        if (!eRequest.has_protocollocation())
+        {
+            eRequest.set_protocollocation(TEST_DEFAULT_PROTOCOL_LOCATION);
+        }
+
+        if (errorMsg.has_value())
+        {
+            return ::api::adapter::genericError<ResponseType>(errorMsg.value());
+        }
+
+        // Set debug mode
+        auto debugMode = [](eTest::DebugMode debugModeValue) -> router::DebugMode
+        {
+            std::unordered_map<eTest::DebugMode, router::DebugMode> debugModeMap = {
+                {eTest::DebugMode::ONLY_OUTPUT, router::DebugMode::ONLY_OUTPUT},
+                {eTest::DebugMode::OUTPUT_AND_TRACES, router::DebugMode::OUTPUT_AND_TRACES},
+                {eTest::DebugMode::OUTPUT_AND_TRACES_WITH_DETAILS, router::DebugMode::OUTPUT_AND_TRACES_WITH_DETAILS}};
+            if (debugModeMap.find(debugModeValue) != debugModeMap.end())
+            {
+                return debugModeMap[debugModeValue];
+            }
+            else
+            {
+                return router::DebugMode::ONLY_OUTPUT;
+            }
+        };
+
+        // Get session
+        auto& sessionManager = SessionManager::getInstance();
+        auto session = sessionManager.getSession(eRequest.name());
+        if (!session.has_value())
+        {
+            return ::api::adapter::genericError<ResponseType>(
+                fmt::format("Session '{}' was not found", eRequest.name()));
+        }
+
+        // Event in Wazuh format
+        auto eventFormat = fmt::format(TEST_EVENT_CONTENT_FORMAT,
+                                       eRequest.protocolqueue(),
+                                       eRequest.protocollocation(),
+                                       eRequest.event().string_value(),
+                                       eRequest.name());
```

Comment:
Please, make variables `const` when possible. The same goes for every possible case.